.github/workflows/bazel.yml

```diff
@@ -85,7 +225,7 @@ jobs:
             bazel-target-rule: "cc_test"
           - bazel-config: "--config=production"
             bazel-target-rule: "cc_test"
-    name: Bazel Build & Test Job
+    name: Bazel Test Job
```

Comment:
Unrelated change: Makes it easier to read the job name (i.g. config type) because the name does not get cut off.